cinema/models.py

```diff
@@ -1,10 +1,36 @@
 from django.db import models
 
 
+class Actor(models.Model):
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return f"{self.first_name}{self.last_name}"
+
+
+class Genre(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=100)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    def __str__(self):
+        return f"{self.name}:(rows:{self.rows}, seats:{self.seats_in_row})"
+
+
 class Movie(models.Model):
     title = models.CharField(max_length=255)
     description = models.TextField()
+    actors = models.ManyToManyField(Actor, related_name='movies')
+    genres = models.ManyToManyField(Genre, related_name="movies")
     duration = models.IntegerField()
 
     def __str__(self):
-        return self.title
+        return f"{self.title}:({self.genres}, {self.duration})"
```

Comment:
In the `__str__` method of the `Movie` model, directly including `self.genres` will not display the genre names as expected. You might want to use a method to join the genre names into a string, such as `', '.join(genre.name for genre in self.genres.all())`.